### ISSUE TITLE
Add bulk actions support to ModelViewSet

### ISSUE THREAD

**Issue by laymonage** [wagtail/wagtail#11371]
### Is your proposal related to a problem?

It would be nice to have the bulk actions feature available to ModelViewSet. It's a functionality that doesn't have to be snippets-specific. We already have it for images, documents, and users, so it would be natural to allow it for arbitrary models.

### Describe the solution you'd like

Have an option in ModelViewSet to enable bulk actions. When enabled, it will automatically add the checkbox column to the index view.

Perhaps also make it so that the actions are registered as part of the viewset, instead of having to use hooks separately?

### Describe alternatives you've considered

Keep bulk actions as a separate feature. Maybe write the docs on how to enable it for ModelViewSet, but leave that for developers to implement.

### Additional context

@Morsey187 managed to get bulk actions to work with ModelViewSet, which looks great as it doesn't seem to require much custom code outside of the custom actions:

https://github.com/wagtail-nest/wagtail-modeladmin/issues/8#issuecomment-1861728726

**Comment by Chiemezuo** [comment]
HI, Sage!
Could I be of any help here?

**Comment by lb-** [comment]
Is it worth looking at the JS implementation as part of this or do you think they'll be best done separately?

One thing to consider is how we set the HTML attributes. My main concern with the current approach is the reliance on setting global values for translations.

I'm hopeful that Stimulus will let us achieve a similar goal but instead set these translations on the root Dom element. On the one that has the controller attached. 

Once https://github.com/wagtail/wagtail/pull/10867 is merged we should have a pretty solid base controller to build on for the rest of the bulk actions migration.

**Comment by thibaudcolas** [comment]
Noting use case from #14073: Bulk delete support for redirects. That’d be a win!
